runtime/oti/phelpers.m4

```diff
@@ -466,27 +501,244 @@ define({SAVE_C_VOLATILE_REGS},{
 	stfd fp5,JIT_FPR_SAVE_SLOT(5)
 	stfd fp6,JIT_FPR_SAVE_SLOT(6)
 	stfd fp7,JIT_FPR_SAVE_SLOT(7)
+ifdef({MUST_PRESERVE_VR},{
+	laddr r4,J9TR_VMThread_javaVM(J9VMTHREAD)
+	lwz r3,J9TR_JavaVM_runtimeFlags(r4)
+	andi. r3,r3,J9TR_J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS
+	beq CONCAT(.L_no_VR_save_,SYM_COUNT)
+	addi r3,r1,J9TR_cframe_jitVRs
+	stxvd2x 0,0,r3
+	addi r3,r3,16
+	stxvd2x 1,0,r3
+	addi r3,r3,16
+	stxvd2x 2,0,r3
+	addi r3,r3,16
+	stxvd2x 3,0,r3
+	addi r3,r3,16
+	stxvd2x 4,0,r3
+	addi r3,r3,16
+	stxvd2x 5,0,r3
+	addi r3,r3,16
+	stxvd2x 6,0,r3
+	addi r3,r3,16
+	stxvd2x 7,0,r3
+	addi r3,r3,16
+	stxvd2x 8,0,r3
+	addi r3,r3,16
+	stxvd2x 9,0,r3
+	addi r3,r3,16
+	stxvd2x 10,0,r3
+	addi r3,r3,16
+	stxvd2x 11,0,r3
+	addi r3,r3,16
+	stxvd2x 12,0,r3
+	addi r3,r3,16
+	stxvd2x 13,0,r3
+	addi r3,r3,16
+	stxvd2x 14,0,r3
+	addi r3,r3,16
+	stxvd2x 15,0,r3
+	addi r3,r3,16
+	stxvd2x 16,0,r3
+	addi r3,r3,16
+	stxvd2x 17,0,r3
+	addi r3,r3,16
+	stxvd2x 18,0,r3
+	addi r3,r3,16
+	stxvd2x 19,0,r3
+	addi r3,r3,16
+	stxvd2x 20,0,r3
+	addi r3,r3,16
+	stxvd2x 21,0,r3
+	addi r3,r3,16
+	stxvd2x 22,0,r3
+	addi r3,r3,16
+	stxvd2x 23,0,r3
+	addi r3,r3,16
+	stxvd2x 24,0,r3
+	addi r3,r3,16
+	stxvd2x 25,0,r3
+	addi r3,r3,16
+	stxvd2x 26,0,r3
+	addi r3,r3,16
+	stxvd2x 27,0,r3
+	addi r3,r3,16
+	stxvd2x 28,0,r3
+	addi r3,r3,16
+	stxvd2x 29,0,r3
+	addi r3,r3,16
+	stxvd2x 30,0,r3
+	addi r3,r3,16
+	stxvd2x 31,0,r3
+	addi r3,r3,16
+	stxvd2x 32,0,r3
+	addi r3,r3,16
+	stxvd2x 33,0,r3
+	addi r3,r3,16
+	stxvd2x 34,0,r3
+	addi r3,r3,16
+	stxvd2x 35,0,r3
+	addi r3,r3,16
+	stxvd2x 36,0,r3
+	addi r3,r3,16
+	stxvd2x 37,0,r3
+	addi r3,r3,16
+	stxvd2x 38,0,r3
+	addi r3,r3,16
+	stxvd2x 39,0,r3
+	addi r3,r3,16
+	stxvd2x 40,0,r3
+	addi r3,r3,16
+	stxvd2x 41,0,r3
+	addi r3,r3,16
+	stxvd2x 42,0,r3
+	addi r3,r3,16
+	stxvd2x 43,0,r3
+	addi r3,r3,16
+	stxvd2x 44,0,r3
+	addi r3,r3,16
+	stxvd2x 45,0,r3
+	addi r3,r3,16
+	stxvd2x 46,0,r3
+	addi r3,r3,16
+	stxvd2x 47,0,r3
+	addi r3,r3,16
+	stxvd2x 48,0,r3
+	addi r3,r3,16
+	stxvd2x 49,0,r3
+	addi r3,r3,16
+	stxvd2x 50,0,r3
+	addi r3,r3,16
+	stxvd2x 51,0,r3
+ifdef({MUST_PRESERVE_FPR},{
+	b CONCAT(.L_no_VR_save_,SYM_COUNT)
```

Comment:
Should not it be save_done? Also, why we are not saving FPR0-FPR7 similar to restoring code below?

Reply:
No, when we don't save VRs we save the other FPRs at the label.

FPR0-7 have been saved above.

Reply:
Sorry, I think you're right - once the VRs are saved, we are in fact done.

Reply:
Fixed.